### PR TITLE
Idaho Capital Gains Deduction

### LINKED ISSUE #2769: Idaho Capital Gain Deduction
reference: https://casetext.com/regulation/idaho-administrative-code/title-idapa-35-tax-commission-state/rule-350101-income-tax-administrative-rules/section-350101170-idaho-capital-gains-deduction-in-general

Tax form: https://tax.idaho.gov/wp-content/uploads/forms/EFO00093/EFO00093_05-19-2022.pdf

### PR DESCRIPTION
**Pull request by rachellin0**
Fixes #2769

### Summary
🛠️📝🚀

This pull request updates the VS Code settings and the changelog for the PolicyEngine US project. It removes the default extension recommendations from `.vscode/extensions.json` and adds a new entry to `changelog.yaml`.

> _`extensions.json`_
> _No suggestions by default_
> _A blank slate for code_

### Walkthrough
*  Add empty recommendations array to VS Code settings ([link](https://github.com/PolicyEngine/policyengine-us/pull/2770/files?diff=unified&w=0#diff-c16655a98a3ee89a7636a59c59a72b0e93649e3a1e947327cfc43a1336b4e912L1-R2))
*  Update changelog with latest update date and time ([link](https://github.com/PolicyEngine/policyengine-us/pull/2770/files?diff=unified&w=0#diff-777b71fb5d4fc10f7c1ceb6a9c0f2430c1a5dd95970d61d71e305934a86c526eR3682))

### CODE REVIEW

File: policyengine_us/parameters/gov/states/id/tax/income/deductions/amount.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+description: IDAHO allows filers to deduct this percentage of their capital gains
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/deductions/amount.yaml:1]:
```suggestion
description: Idaho allows filers to deduct this percentage of their capital gains.
```

File: policyengine_us/parameters/gov/states/id/tax/income/deductions/amount.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+values:
+  2022-01-01: 0.6
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/deductions/amount.yaml:3]: need a 2021 value + 2021 reference

File: policyengine_us/parameters/gov/states/id/tax/income/deductions/amount.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+metadata:
+  unit: /1
+  label: IDAHO CAPITAL GAINS DEDUCTION element percent
+  reference:
+  - title: 2022 IDAHO Tax form - Capital Gains Deduction
+    href: https://tax.idaho.gov/wp-content/uploads/forms/EFO00093/EFO00093_05-19-2022.pdf
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/deductions/amount.yaml:9]: add page number

File: policyengine_us/parameters/gov/states/id/tax/income/deductions/amount.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  - title: IDAHO CAPITAL GAINS DEDUCTION-Section 35.01.01.170
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/deductions/amount.yaml:10]: add specific subsection

File: policyengine_us/variables/gov/states/id/tax/income/deductions/id_capital_gains_deduction.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        decuctions = p.amount * capital_gains
+
+        # capital gains deduction may not exceed the capital gain net income included in taxable income
+        qualified_deductions = min_(capital_gain_net_income, decuctions)
+
+        # Ordinary Income do not qualify for the Idaho capital gains deduction
+        # Gain from dispositions of certain depreciable property treated as ordinary income
+
+        # Return calculated amount.
+        return qualified_deductions
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/id/tax/income/deductions/id_capital_gains_deduction.py:36]:
```suggestion
```

File: policyengine_us/variables/gov/states/id/tax/income/deductions/id_capital_gains_deduction.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    definition_period = YEAR
+    reference = "https://casetext.com/regulation/idaho-administrative-code/title-idapa-35-tax-commission-state/rule-350101-income-tax-administrative-rules/section-350101170-idaho-capital-gains-deduction-in-general"
+    defined_for = StateCode.ID
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(
+            period
+        ).gov.states.id.tax.income.deductions.capital_gains
+        # taxpayer must report capital gain net income
+        capital_gains = max_(0, add(tax_unit, period, ["capital_gains"]))
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/id/tax/income/deductions/id_capital_gains_deduction.py:18]:
```suggestion
        capital_gains = add(tax_unit, period, ["capital_gains"])
```

File: policyengine_us/variables/gov/states/id/tax/income/deductions/id_capital_gains_deduction.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+        # capital_gain_net_income
+
+        capital_loss = tax_unit(
+            "capital_losses", period
+        )  # Idaho capital gains deduction may not be netted against gains from property qualifying for the Idaho capital gains deduction before the amount of the deduction is determined
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/id/tax/income/deductions/id_capital_gains_deduction.py:24]: split comment into 2 lines - need to make it shorter

File: policyengine_us/parameters/gov/states/id/tax/income/deductions/amount.yaml [hunk 8]
```diff
@@ -0,0 +1,14 @@
+description: Idaho allows filers to deduct this percentage of their capital gains
+values:
+  2022-01-01: 0.6
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/deductions/amount.yaml:3]:
```suggestion
```

File: policyengine_us/parameters/gov/states/id/tax/income/deductions/amount.yaml [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+  2021-01-01: 0.6
+metadata:
+  unit: /1
+  label: IDAHO CAPITAL GAINS DEDUCTION element percent
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/deductions/amount.yaml:7]:
```suggestion
  label: Idaho capital gains deduction percentage
```

File: policyengine_us/parameters/gov/states/id/tax/income/deductions/amount.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/deductions/amount.yaml:1]: rename to percentage.yaml

File: policyengine_us/parameters/gov/states/id/tax/income/deductions/amount.yaml [hunk 10]
```diff
@@ -0,0 +1,14 @@
+description: Idaho allows filers to deduct this percentage of their capital gains
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/deductions/amount.yaml:1]:
```suggestion
description: Idaho allows filers to deduct this percentage of their capital gains.
```

File: policyengine_us/parameters/gov/states/id/tax/income/deductions/amount.yaml [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+values:
+  2022-01-01: 0.6
+  2021-01-01: 0.6
+metadata:
+  unit: /1
+  label: IDAHO CAPITAL GAINS DEDUCTION element percent
+  reference:
+  - title: 2022 Idaho Tax form - Capital Gains Deduction-page 1
+    href: https://tax.idaho.gov/wp-content/uploads/forms/EFO00093/EFO00093_05-19-2022.pdf
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/deductions/amount.yaml:10]: add page to pdf

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: need changelog

File: policyengine_us/parameters/gov/states/id/tax/income/deductions/capital_gains/percentage.yaml [hunk 12]
```diff
@@ -0,0 +1,13 @@
+description: Idaho allows filers to deduct this percentage of their capital gains.
+values:
+  2021-01-01: 0.6
+metadata:
+  unit: /1
+  label: IDAHO CAPITAL GAINS DEDUCTION element percent
+  reference:
+  - title: 2022 Idaho Tax form - Capital Gains Deduction-pg. 1
+    href: https://tax.idaho.gov/wp-content/uploads/forms/EFO00093/EFO00093_05-19-2022.pdf
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/deductions/capital_gains/percentage.yaml:9]: need pdf page in link

File: policyengine_us/parameters/gov/states/id/tax/income/deductions/capital_gains/percentage.yaml [hunk 13]
```diff
@@ -0,0 +1,13 @@
...
+  - title: IDAHO CAPITAL GAINS DEDUCTION-Section 35.01.01.170- pg.1- 01.02
+    href: https://casetext.com/regulation/idaho-administrative-code/title-idapa-35-tax-commission-state/rule-350101-income-tax-administrative-rules/section-350101170-idaho-capital-gains-deduction-in-general
+  - title: 2021 Idaho Tax form - Capital Gains Deduction-pg. 1
+    href: https://tax.idaho.gov/wp-content/uploads/forms/EFO00093/EFO00093_09-15-2021.pdf
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/deductions/capital_gains/percentage.yaml:13]: need pdf page in link

File: policyengine_us/tests/policy/baseline/gov/states/id/tax/income/deductions/id_capital_gains_deduction.yaml
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/id/tax/income/deductions/id_capital_gains_deduction.yaml:1]: test edge cases: losses > gains f.e.

File: policyengine_us/parameters/gov/states/id/tax/income/deductions/capital_gains/percentage.yaml [hunk 14]
```diff
@@ -0,0 +1,13 @@
+description: Idaho allows filers to deduct this percentage of their capital gains.
+values:
+  2021-01-01: 0.6
+metadata:
+  unit: /1
+  label: IDAHO CAPITAL GAINS DEDUCTION element percent
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/deductions/capital_gains/percentage.yaml:6]:
```suggestion
  label: Idaho capital gains deduction percentage
```

File: policyengine_us/variables/gov/states/id/tax/income/deductions/id_capital_gains_deduction.py [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+    def formula(tax_unit, period, parameters):
+        person = tax_unit.members
+        p = parameters(
+            period
+        ).gov.states.id.tax.income.deductions.capital_gains
+        # taxpayer must report capital gain net income
+        capital_gains = add(tax_unit, period, ["capital_gains"])
+
+        # capital_gain_net_income
+        capital_loss = person("capital_losses", period)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/id/tax/income/deductions/id_capital_gains_deduction.py:22]: use `net_capital_gain` instead of gains - losses

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/id/tax/income/deductions/id_capital_gains_deduction.py:22]: should be more straight forward

File: policyengine_us/parameters/gov/states/id/tax/income/deductions/capital_gains/percentage.yaml [hunk 16]
```diff
@@ -0,0 +1,13 @@
+description: Idaho allows filers to deduct this percentage of their capital gains.
+values:
+  2021-01-01: 0.6
+metadata:
+  unit: /1
+  label: Idaho capital gains deduction percentage
+  reference:
+  - title: 2022 Idaho Tax form - Capital Gains Deduction-pg. 1
+    href: https://tax.idaho.gov/wp-content/uploads/forms/EFO00093/EFO00093_05-19-2022.pdf-pg. 1
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/deductions/capital_gains/percentage.yaml:9]:
```suggestion
    href: https://tax.idaho.gov/wp-content/uploads/forms/EFO00093/EFO00093_05-19-2022.pdf#page=1
```

File: policyengine_us/parameters/gov/states/id/tax/income/deductions/capital_gains/percentage.yaml [hunk 17]
```diff
@@ -0,0 +1,13 @@
...
+  - title: IDAHO CAPITAL GAINS DEDUCTION-Section 35.01.01.170- pg.1- 01.02
+    href: https://casetext.com/regulation/idaho-administrative-code/title-idapa-35-tax-commission-state/rule-350101-income-tax-administrative-rules/section-350101170-idaho-capital-gains-deduction-in-general
+  - title: 2021 Idaho Tax form - Capital Gains Deduction-pg. 1
+    href: https://tax.idaho.gov/wp-content/uploads/forms/EFO00093/EFO00093_09-15-2021.pdf-pg. 1
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/deductions/capital_gains/percentage.yaml:13]:
```suggestion
    href: https://tax.idaho.gov/wp-content/uploads/forms/EFO00093/EFO00093_09-15-2021.pdf#page=1
```

File: policyengine_us/parameters/gov/states/id/tax/income/deductions/capital_gains/percentage.yaml [hunk 18]
```diff
@@ -0,0 +1,13 @@
...
+  2021-01-01: 0.6
+metadata:
+  unit: /1
+  label: Idaho capital gains deduction percentage
+  reference:
+  - title: 2022 Idaho Tax form - Capital Gains Deduction-pg. 1
+    href: https://tax.idaho.gov/wp-content/uploads/forms/EFO00093/EFO00093_05-19-2022.pdf-pg. 1
+  - title: IDAHO CAPITAL GAINS DEDUCTION-Section 35.01.01.170- pg.1- 01.02
+    href: https://casetext.com/regulation/idaho-administrative-code/title-idapa-35-tax-commission-state/rule-350101-income-tax-administrative-rules/section-350101170-idaho-capital-gains-deduction-in-general
+  - title: 2021 Idaho Tax form - Capital Gains Deduction-pg. 1
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/deductions/capital_gains/percentage.yaml:12]:
```suggestion
  - title: 2021 Idaho Tax form - Capital Gains Deduction
```

File: policyengine_us/parameters/gov/states/id/tax/income/deductions/capital_gains/percentage.yaml [hunk 19]
```diff
@@ -0,0 +1,13 @@
+description: Idaho allows filers to deduct this percentage of their capital gains.
+values:
+  2021-01-01: 0.6
+metadata:
+  unit: /1
+  label: Idaho capital gains deduction percentage
+  reference:
+  - title: 2022 Idaho Tax form - Capital Gains Deduction-pg. 1
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/deductions/capital_gains/percentage.yaml:8]:
```suggestion
  - title: 2022 Idaho Tax form - Capital Gains Deduction
```

File: policyengine_us/parameters/gov/states/id/tax/income/deductions/capital_gains/percentage.yaml [hunk 20]
```diff
@@ -0,0 +1,13 @@
...
+    href: https://tax.idaho.gov/wp-content/uploads/forms/EFO00093/EFO00093_05-19-2022.pdf-pg. 1
+  - title: IDAHO CAPITAL GAINS DEDUCTION-Section 35.01.01.170- pg.1- 01.02
```
Comment (qingqiuzhang) [inline policyengine_us/parameters/gov/states/id/tax/income/deductions/capital_gains/percentage.yaml:10]:
```suggestion
  - title: Section 35.01.01.170 - IDAHO CAPITAL GAINS DEDUCTION
```

File: policyengine_us/tests/policy/baseline/gov/states/id/tax/income/deductions/id_capital_gains_deduction.yaml [hunk 21]
```diff
@@ -0,0 +1,17 @@
+- name: No capital gains
```
Comment (qingqiuzhang) [inline policyengine_us/tests/policy/baseline/gov/states/id/tax/income/deductions/id_capital_gains_deduction.yaml:1]:
```suggestion
- name: Filers net capital gain = 0, Idaho capital gains deduction = 0
```

File: policyengine_us/tests/policy/baseline/gov/states/id/tax/income/deductions/id_capital_gains_deduction.yaml [hunk 22]
```diff
@@ -0,0 +1,17 @@
...
+  period: 2022
+  absolute_error_margin: 0.001
+  input:
+    net_capital_gain: 0
+    state_code: ID
+  output:
+    id_capital_gains_deduction: 0
+
+- name: 60% net capital gains
```
Comment (qingqiuzhang) [inline policyengine_us/tests/policy/baseline/gov/states/id/tax/income/deductions/id_capital_gains_deduction.yaml:10]:
```suggestion
- name: Filers net capital gain = 6_000, Idaho capital gains deduction = 3_600.

**Pull request closed without merging**